core/blockchain.go

```diff
@@ -1555,6 +1555,16 @@ func (bc *BlockChain) writeKnownBlock(block *types.Block) error {
 	return nil
 }
 
+func (bc *BlockChain) TimeBeforeFlush() (time.Duration, error) {
```

Comment:
nitpick: I think we should rename the method to something like `ProcTimeBeforeFlush` as the returned duration is not absolute time but the processing time accumulated in gcproc. Gcproc is a hugely different measure e.g. 1 h of proc time can take ~12 hours of real time.